### DISCUSSION TITLE
Weird Transparent Borders Around Popups in Certain Applications

### DISCUSSION THREAD

**Discussion by auralisx** [hyprwm/Hyprland#7509] [Q&A] [answered]
I've been experiencing an issue with transparent borders around popups in certain applications like Brave, Edge, and VSCode. The issue is most noticeable with popup menus and similar UI elements. In VSCode, using the custom title bar style seems to resolve the issue, but the transparent borders persist when using the native title bar style.

After some troubleshooting, I've identified that the blur setting appears to be the cause. When I disable the blur, the transparent borders disappear across all these applications.

Has anyone else encountered this issue? Are there any known workarounds or fixes that could help resolve this while keeping the blur effect enabled?

Thanks in advance for any suggestions!
![image](https://github.com/user-attachments/assets/1e82cc6d-1ce2-4860-93bd-93a61c55064b)
![image](https://github.com/user-attachments/assets/1ee1c8ab-50e0-47fd-a33c-8b40aeeee713)

**Comment by vaxerski** [comment, marked as answer]
xwayland limitation, just launch your browser in native wayland mode

**Reply by auralisx** [reply]
Thanks for the suggestion! Launching the browser in native Wayland mode with flags seems to have resolved the issue somewhat. there is still some shadow appearing but it is better than those thick transparent borders. 

As a follow-up, I wanted to ask if there’s a way to set environment variables like  [ELECTRON_OZONE_PLATFORM_HINT=auto or wayland](https://www.electronjs.org/docs/latest/api/environment-variables#electron_ozone_platform_hint-linux)  for Electron-based applications directly within the Hyprland configuration with environment variables.

**Reply by vaxerski** [reply]
`env = ENV,value`
